clientv3/integration/lease_test.go

```diff
@@ -261,12 +256,12 @@ func TestLeaseGrantErrConnClosed(t *testing.T) {
 	defer clus.Terminate(t)
 
 	cli := clus.Client(0)
-	le := clientv3.NewLease(cli)
+	clus.TakeClient(0)
 
 	donec := make(chan struct{})
 	go func() {
 		defer close(donec)
-		_, err := le.Grant(context.TODO(), 5)
+		_, err := cli.Grant(context.TODO(), 5)
 		if err != nil && err != grpc.ErrClientConnClosing {
```

Comment:
I think it's related to `cancelWhenStop`; the client close will trigger a different path than what was there before. Probably shouldn't `cancelWhenStop` at all, it's really old code that doesn't reflect how the client operates now.